Tests/Unit/Hooks/HookProviderTest.php

```diff
@@ -0,0 +1,146 @@
+     */
+
+    /**
+     * Creates a TestHook implementor object.
+     *
+     * @return TestHookImplementor
+     */
+    private function createTestHookImplementor(): TestHookImplementor
+    {
+        return GeneralUtility::makeInstance(TestHookImplementor::class);
+    }
+
+    /**
+     * Creates a TestHook accepting Hook object.
+     *
+     * @return HookProvider
+     */
+    private function createHookObject(): HookProvider
+    {
+        return GeneralUtility::makeInstance(HookProvider::class, TestHookInterface::class);
+    }
+
+    /*
+     * Tests concerning the TestHookImplementor
+     */
+
+    /**
+     * @test
+     */
+    public function hookImplementorCanBeCreated()
+    {
+        self::assertInstanceOf(TestHookImplementor::class, $this->createTestHookImplementor());
+    }
+
+    /**
+     * @test
+     */
+    public function hookImplementorImplementsHookHierachy()
+    {
+        $implementor = $this->createTestHookImplementor();
+
+        self::assertInstanceOf(Hook::class, $implementor);
+
+        self::assertInstanceOf(TestHookInterface::class, $implementor);
+    }
+
+    /**
+     * @test
+     */
+    public function hookImplementorTestHookMethodCanBeCalledAndReportsBeingCalled()
+    {
+        $implementor = $this->createTestHookImplementor();
+
+        $implementor->testHookMethod();
+
+        self::assertTrue($implementor->getWasCalled());
+    }
+
+    /*
+     * Tests concerning the Hook object
+     */
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookCanBeCreated()
+    {
+        self::assertInstanceOf(HookProvider::class, $this->createHookObject());
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithNoHookImplemetorRegisteredResultsInEmptyHookList()
+    {
+        unset($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestHookInterface::class]);
+        $hookObject = $this->createHookObject();
+
+        self::assertEmpty($hookObject->getHooks());
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithOneHookImplemetorRegisteredResultsInOneHookInHookList()
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestHookInterface::class][1565007112] =
+            TestHookImplementor::class;
+        $hookObject = $this->createHookObject();
+
+        self::assertCount(1, $hookObject->getHooks());
+        self::assertContainsOnlyInstancesOf(TestHookInterface::class, $hookObject->getHooks());
+        self::assertContainsOnlyInstancesOf(TestHookImplementor::class, $hookObject->getHooks());
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithTwoHookImplemetorsRegisteredResultsInTwoHooksInHookList()
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestHookInterface::class][1565007112] =
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestHookInterface::class][1565007113] =
+            TestHookImplementor::class;
```

Comment:
Please use two statement for this, and please use two different hook implementors. (Adding the same implementor twice is not supported as far as I understand the concept of hooks.)